spec/lib/pedicel/base_spec.rb

```diff
@@ -233,16 +233,15 @@
 
       it 'handles that one certificate can be both intermediate and leaf' do
         def create_special_certificate(ca_key, ca_certificate, config, oids)
-          key = OpenSSL::PKey::EC.new(PedicelPay::EC_CURVE)
-          key.generate_key
+          key = OpenSSL::PKey::EC.generate(PedicelPay::EC_CURVE)
 
           cert = OpenSSL::X509::Certificate.new
           # https://www.ietf.org/rfc/rfc5280.txt -> Section 4.1, search for "v3(2)".
           cert.version = 2
           cert.serial = 1
           cert.subject = config[:subject][:intermediate]
           cert.issuer = ca_certificate.subject
-          cert.public_key = PedicelPay::Helper.ec_key_to_pkey_public_key(key)
+          cert.public_key = key # PedicelPay::Helper.ec_key_to_pkey_public_key(key)
```

Comment:
```suggestion
          cert.public_key = key
```

Reply:
In a future PR on pedicel-pay we're considering dropping the `PedicelPay::Helper.ec_key_to_pkey_public_key` method as the security (dropping the private key) isn't strictly necessary.

Reply:
Done. It seems like I forgot to uncomment this one, but I guess it's fine if you are going to remove that method anyway.
If we are already discussing this, do you want me to remove this method already in my Pedicel-Pay PR?

Reply:
@cub8 We would actually like you to do exactly that :+1: :bow: 

Less crypto is better when it's not strictly necessary, and it would be nice to have a similar approach in pedicel(-pay) and aliquot(-pay) (see https://github.com/clearhaus/aliquot-pay/pull/12 even though its title is incorrect).

Reply:
Uhm, hold your horses. The method `PedicelPay::Helper.ec_key_to_pkey_public_key` is used by specs in pedicel, and I'm not currently sure how to avoid using it there.

Reply:
You would have to use `OpenSSL::PKey::EC::Point` to build `OpenSSL::PKey::EC` instance.
In `pedicel-pay` there are also situation when `OpenSSL::PKey::EC::Point` is used to build `OpenSSL::PKey::EC` instance, i.e. `token_header.rb:20` or `backend.rb:93`.

Reply:
I'm becoming more and more convinced that the solution via ASN1 is necessary because we want to support both Ruby 2.7 and 3.x. I don't see a simple way to come from an EC point to the public key in pedicel-pay's `token_header.rb:20` and `backend.rb:93`. Well, `EC::Point#public_to_(der|pem)` is simple, but we only have that in Ruby 3.x.

@cub8 Am I missing something? Do you perhaps agree?

Reply:
On another note it seems we can remove the unused method in pedicel's `spec/spec_helper.rb`.

Reply:
`OpenSSL::PKey::EC::Point` doesn't have `public_to(der|pem)` methods. `OpenSSL::PKey::EC` does in Ruby 3.x.

I don't know if there's any other workaround for our issue than ASN1 solution. However, I am not that familiar with OpenSSL library, so maybe someone else will come up with a solution.  Right now, the ASN1 one works well, so I think it should be kept if there's nothing better.

I will remove unused method right away.

Reply:
Let's keep the ASN1 solution then :+1: